packages/nextra/src/setup-page.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * ⚠️ Attention!
+ * This file should be never used directly, only in loader.ts
+ */
+
+import { FC } from 'react'
+import get from 'lodash.get'
+import { IS_PRODUCTION, NEXTRA_INTERNAL } from './constants'
+import {
+  DynamicMetaDescriptor,
+  NextraInternalGlobal,
+  PageOpts,
+  ThemeConfig
+} from './types'
+
+/**
+ * Calculate a 32 bit FNV-1a hash
+ * Found here: https://gist.github.com/vaiorabbit/5657561
+ * Ref.: http://isthe.com/chongo/tech/comp/fnv/
+ *
+ * @param {string} str the input value
+ * @param {number} [seed] optionally pass the hash of the previous chunk
+ * @returns {string}
+ */
+function hashFnv32a(str: string, seed = 0x811c9dc5): string {
+  let hval = seed
+
+  for (let i = 0; i < str.length; i++) {
+    hval ^= str.charCodeAt(i)
+    hval += (hval << 1) + (hval << 4) + (hval << 7) + (hval << 8) + (hval << 24)
+  }
+
+  // Convert to 8 digit hex string
+  return ('0000000' + (hval >>> 0).toString(16)).substring(-8)
+}
+
+export function setupNextraPage({
```

Comment:
we can choose a better name

Reply:
Thanks, love putting loader content inside another file, makes things super clean and typed! One thing I'm not sure is all the content here (`lodash.get`, `hashFnv32a`) will be part of the page bundle that make the page larger. They should be done at build time I think. Maybe `hashFnv32a(JSON.stringify(pageOpts))` can be passed inline?

```js
setupNextraPage({
  pageOptsHash: "${pageOptsChecksum}"
})
```

Another thing is probably Webpack's dead code elimination problem, using `if (!IS_PRODUCTION && hot)` and `if (process.env.NODE_ENV !== 'production' && hot)` could be different. The former needs to actually run the code to tell if that condition will be executed or not, but the latter can be determined when building the module via static code analyzation so that code will be removed completely after build.

Reply:
you are right, after adding `process.env.NODE_ENV !== 'production'` or `process.env.NODE_ENV === 'development'` `.next/static/chunks` folder doesn't include `module.hot` code, and `hashFnv32a` is automatically tree-shaked, and about `lodash.get` let's keep for the moment

Reply:
before/after
<img width="566" alt="image" src="https://user-images.githubusercontent.com/7361780/212991993-6192bea3-b3cb-4570-98f8-9776db48dc5f.png">